dotnet/src/Skills/Skills.OpenAPI/Model/RestApiOperation.cs

```diff
@@ -105,32 +105,38 @@ public sealed class RestApiOperation
     /// Builds operation Url.
     /// </summary>
     /// <param name="arguments">The operation arguments.</param>
+    /// <param name="serverUrlOverride">Override for REST API operation server url.</param>
+    /// <param name="documentUri">The URI of OpenApi document.</param>
     /// <returns>The operation Url.</returns>
-    public Uri BuildOperationUrl(IDictionary<string, string> arguments)
+    public Uri BuildOperationUrl(IDictionary<string, string> arguments, Uri? serverUrlOverride = null, Uri? documentUri = null)
     {
         var path = this.ReplacePathParameters(this.Path, arguments);
 
         path = this.AddQueryString(path, arguments);
 
-        Uri serverUrl;
+        string serverUrl;
 
-        //Override defined server url - https://api.example.com/v1 by the one from arguments.
-        if (arguments.TryGetValue(ServerUrlArgumentName, out string serverUrlString))
+        if (serverUrlOverride is not null)
```

Comment:
I did it on purpose, but in order to explain why it's needed, long text is required :)

In this functionality we have 3 sources, from where we can get `Server URL` (sorted by priority):
- `ServerUrlOverride` property: it's part of `OpenApiSkillExecutionParameters` parameters and it has top priority. As soon as user set this field programmatically, it will be always used.
- First item in `servers` property of OpenAPI document: this is used when `ServerUrlOverride` property is null.
- OpenAPI document host URL: it's used when `ServerUrlOverride` is null and there are no values in `servers` property of OpenAPI document - basically the feature, which allows to use relative URLs, main purpose of this PR.

Now, as soon as we get `Server URL` from any of the sources above - we have to set it in two places:
- Context variable `server-url`: we send it to LLM as parameter, so LLM can decide whether it's needed to be passed to the next function in pipeline or not (for example in planning functionality).
- In `HttpClient` to make an actual call to endpoint described in OpenAPI document. In this scenario there is also another source of `Server URL` - context variable, which is described in previous point, and the value at this stage can be the same as it was when we set it or it can be updated by LLM for example, if needed.

In case of saving this value in context variable - the value is not mandatory, because LLM can decide that it's not needed at all. That's why `RestApiOperationExtensions.GetServerUrlParameter` doesn't throw an exception and we don't need logging there, because this scenario is acceptable. 
While for second case, where we need to set it in `HttpClient` to make an actual call to endpoint, `Server URL` is required, otherwise it won't be possible to make any calls, which means that we won't be able to execute a function - in this case we should throw an exception.

Taking into account both scenarios, the logic for getting server URL can't be re-used in both places, since in one place it's acceptable to have `null` value, while in other place it's mandatory and we should throw an exception. The other difference is that in case of context variable we have 3 sources of `Server URL` parameter (`ServerUrlOverride`, `servers` or host URL) while in second case we have 4 sources (`ServerUrlOverride`, context variable, `servers` or host URL).

Let me know if this explains the implementation or more details needed.

Reply:
UPD: I removed helper extension method, but explanation above should still provide some context for this functionality.

Reply:
Resolving this comment, since extension method doesn't exist anymore. Thanks!